### ISSUE TITLE
Reader does not return any message when configured with start with inclusive latest messageId 

### ISSUE THREAD

**Issue by tomjohn9** [apache/pulsar#10483]
**Describe the bug**
Creating reader with messageId pointing to `MessageId.latest` together with `startMessageIdInclusive`,there is no message returned when `readNext` is invoked. Newly coming messages are consumed. When using `MessageId.earliest`, a message is returned - means there are messages in that topic.

Having running local docker image with puslar 2.7.1 and java client 2.7.1

**Steps To Reproduce**

```
pulsarClient.newReader(JSONSchema.of(MySchema.class))
                .topic(topic)
                .startMessageId(MessageId.latest)
                .startMessageIdInclusive().create()
                .readNext(250, TimeUnit.MILLISECONDS)
```

**Expected behavior**
Last message in queue is returned when calling `Reader.readNext` with Reader configuration` MessageId.latest` and `startMessageIdInclusive()`

**Similar Issue**
Seems, it was already reported in [issue 4912](https://github.com/apache/pulsar/issues/4912) in first paragraph **Test3** 

> Test3:When configuring Reader with startMessageIdInclusive and positioning stream at MessageId.latest, then the next call to Reader.readNext(timeout) should have returned the last message of the topic, but it times out and returns null (without a configured timeout it blocks forever).
.....
Expected behavior
test3 expected to return the last message in topic, test4 hasMessageAvailable expected to be false.

 but it is little bit confusing because underlying method `readMessage` starts with `hasMessageAvailable()` which in that time returns wrongly true and `readNext` is reached but no message is returned, but should be.

**Comment by PatrykWitkowski** [comment]
If you invoke `reader.hasMessageAvailable()` before `reader.readNext()` then it works as expected.
It looks like in that case during searching if there're any messages available a message id for the last message is found and the reader pointer is set for it.
Logs:
```
17:14:42.290 [] [pulsar-client-io-1-1] INFO Get topic last message Id
17:14:42.290 [] [pulsar-client-io-1-1] INFO Successfully getLastMessageId 150:0
17:14:42.293 [] [pulsar-client-io-1-1] INFO Seek subscription to message id 150:0:-1
17:14:42.308 [] [pulsar-client-io-1-1] INFO Successfully reset subscription to message id 150:0:-1
```

You can use it as a **workaround**, but still this issue should be fixed.

**Comment by adrianog** [comment]
Yes and please can startMessageIdInclusive be made available to c++/python as well. It is a glaring omission. #10328

**Comment by codelipenghui** [comment]
Here is the discussion thread https://lists.apache.org/thread/kwljdckm0ykstq8oodz6550hqxdltc98 for this issue.

**Comment by tsturzl** [comment]
This is still an issue, also the work around fails for me as the reader eventually fails when it reaches the end of the topic.

I publish a message on a topic, I start a reader with latest or earliest and startMessageIdInclusive on. I read the first message, I get the message I expect, but then on the next read I get the following.

```
2024-03-19 17:14:36:409% [pulsar-client-io-1-5] ERROR org.apache.pulsar.client.impl.ConsumerImpl - [non-persistent://public/testing/new.topic][subscription-1234567890--reader-60269af7b2] Failed getLastMessageId command

The subscription subscription-1234567890-reader-60269af7b2 of the topic non-persistent://public/testing/new.topic gets the last message id was failed
```
